src/Compilers/Core/Portable/DiagnosticAnalyzer/CachingSemanticModelProvider.cs

```diff
@@ -23,30 +23,31 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal sealed class CachingSemanticModelProvider : SemanticModelProvider
     {
+        public static CachingSemanticModelProvider Instance { get; } = new CachingSemanticModelProvider();
+
         private static readonly ConditionalWeakTable<Compilation, PerCompilationProvider>.CreateValueCallback s_createProviderCallback
             = new ConditionalWeakTable<Compilation, PerCompilationProvider>.CreateValueCallback(compilation => new PerCompilationProvider(compilation));
 
-        private readonly ConditionalWeakTable<Compilation, PerCompilationProvider> _providerCache;
+        private static readonly ConditionalWeakTable<Compilation, PerCompilationProvider> s_providerCache = new ConditionalWeakTable<Compilation, PerCompilationProvider>();
```

Comment:
Why was this made `static`? Couldn't it remain an instance field with the same effect since the owner is a singleton now?

Reply:
The general guidance per @DustinCampbell is that CWT declarations should be static if possible to remove the cyclic leak possibility. But yes, it doesn't *need* to be static now that the owner is a singleton, but I figured it was best to do so.